seaworthy/containers/base.py

```diff
@@ -115,15 +115,54 @@ def clean(self):
         """
         raise NotImplementedError()
 
-    def get_host_port(self, port_spec, index=0):
+    @property
+    def ports(self):
         """
-        :param port_spec: A container port mapping specifier.
+        The ports (exposed and published) of the container.
+        """
+        return self.inner().attrs['NetworkSettings']['Ports']
+
+    def _host_port(self, port_spec, index):
+        if port_spec not in self.ports:
+            raise ValueError("Port '{}' is not exposed".format(port_spec))
+
+        mappings = self.ports[port_spec]
+        if mappings is None:
+            raise ValueError(
+                "Port '{}' is not published to the host".format(port_spec))
+
+        mapping = mappings[index]
+        return mapping['HostIp'], mapping['HostPort']
```

Comment:
Maybe it'd be nice to turn the IP into an [`ipaddress`](https://docs.python.org/3/library/ipaddress.html)?

Reply:
Unless we're happy that it won't break all the things expecting a string (like wedging it into a URL or whatever) it's probably best to leave it for now.